lib/body.js

```diff
@@ -0,0 +1,120 @@
+import {EventEmitter2} from 'eventemitter2';
+import {Message} from 'roslib';
+
+class Body extends EventEmitter2 {
+  constructor(robot) {
+    super();
+
+    const ros = robot.ros;
+
+    // Define the action
+    this.action = new ROSLIB.ActionClient({
+      ros: ros,
+      serverName: 'body/joint_trajectory_action',
+      actionName: 'control_msgs/FollowJointTrajectoryAction',
+      timeout: 10,
+    });
+
+    // Get the skills parameters from the parameter server to extract the
+    // possible motions.
+    // ToDo: get rid of hardcoded "amigo"
+    var param = new ROSLIB.Param({
+      ros: ros,
+      name: '/amigo/skills'
+    });
+
+    var that = this
+    param.get(function(param) {
+      that.joint_names = {}
+      that.default_configurations = {}
+      that.joint_names["torso"] = param.torso.joint_names
+      that.default_configurations["torso"] = param.torso.default_configurations
+      that.default_configurations["left_arm"] = param.arm.default_configurations
+      that.default_configurations["right_arm"] = param.arm.default_configurations
+      that.joint_names["left_arm"] = []
+      that.joint_names["right_arm"] = []
+      param.arm.joint_names.forEach(function(e) {
+        that.joint_names["left_arm"].push(e + "_left")
+        that.joint_names["right_arm"].push(e + "_right")
+      })
+
+    });
+
+    // Define the gripper action
+    this.left_gripper_action = new ROSLIB.ActionClient({
+      ros: ros,
+      serverName: 'left_arm/gripper/action',
+      actionName: 'tue_manipulation_msgs/GripperCommandAction',
+      timeout: 0.0,
+    });
+
+    // Define the gripper action
+    this.right_gripper_action = new ROSLIB.ActionClient({
+      ros: ros,
+      serverName: 'right_arm/gripper/action',
+      actionName: 'tue_manipulation_msgs/GripperCommandAction',
+      timeout: 0.0,
+    });
+
+  }  // End of constructor
+
+  sendGoal(cmd) {
```

Comment:
Should we replace this with sendJointGoal? Might be more convenient if we ever choose to add Cartesian goals as well